<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Курсовой Проект</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>7</x>
      <y>10</y>
      <width>781</width>
      <height>531</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0" colspan="4">
      <widget class="QLabel" name="lTerm">
       <property name="text">
        <string>Алфавит:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0" colspan="4">
      <widget class="QLineEdit" name="leTerm">
       <property name="tabletTracking">
        <bool>false</bool>
       </property>
       <property name="toolTip">
        <string>Входной алфавит</string>
       </property>
       <property name="inputMask">
        <string/>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="maxLength">
        <number>32767</number>
       </property>
       <property name="cursorPosition">
        <number>0</number>
       </property>
       <property name="placeholderText">
        <string>a b c</string>
       </property>
      </widget>
     </item>
     <item row="0" column="4">
      <widget class="QLabel" name="lRegExp">
       <property name="text">
        <string>Регулярное выражение:</string>
       </property>
      </widget>
     </item>
     <item row="5" column="0" colspan="4">
      <widget class="QLineEdit" name="leSubString">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="toolTip">
        <string>Подстрока, которая обязательно должна встречаться</string>
       </property>
       <property name="placeholderText">
        <string>bc</string>
       </property>
      </widget>
     </item>
     <item row="3" column="3">
      <widget class="QSpinBox" name="sbMul">
       <property name="toolTip">
        <string>Кратность вхождения выбранного символа</string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>5</number>
       </property>
      </widget>
     </item>
     <item row="7" column="0" colspan="4">
      <widget class="Line" name="line">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="9" column="0" rowspan="2">
      <widget class="QLabel" name="lFrom">
       <property name="text">
        <string>От:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0" colspan="4">
      <widget class="QLabel" name="lRepeat">
       <property name="text">
        <string>Кратность вхождения некоторого символа:</string>
       </property>
      </widget>
     </item>
     <item row="6" column="0" colspan="4">
      <widget class="QPushButton" name="pbCreateReg">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="toolTip">
        <string>Создает регулярное выражение на основе введенных кратности и подстроки</string>
       </property>
       <property name="text">
        <string>Создать Регулярное Выражение</string>
       </property>
      </widget>
     </item>
     <item row="9" column="2" rowspan="2">
      <widget class="QLabel" name="lTo">
       <property name="text">
        <string>До:</string>
       </property>
      </widget>
     </item>
     <item row="9" column="3" rowspan="2">
      <widget class="QSpinBox" name="sbTo">
       <property name="toolTip">
        <string>Максимальная длина цепочек на вывод</string>
       </property>
       <property name="suffix">
        <string/>
       </property>
       <property name="maximum">
        <number>6</number>
       </property>
       <property name="value">
        <number>6</number>
       </property>
      </widget>
     </item>
     <item row="11" column="0" colspan="4">
      <widget class="QPushButton" name="pbOutput">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="toolTip">
        <string>Выводит цепочки заданой длины на основе регулярного выражения</string>
       </property>
       <property name="text">
        <string>Вывод</string>
       </property>
      </widget>
     </item>
     <item row="8" column="0" colspan="4">
      <widget class="QLabel" name="lOutPut">
       <property name="text">
        <string>Вывод цепочек заданной длины:</string>
       </property>
      </widget>
     </item>
     <item row="4" column="0" colspan="4">
      <widget class="QLabel" name="lSubString">
       <property name="text">
        <string>Подстрока:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0" colspan="3">
      <widget class="QComboBox" name="cbRepeat">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="toolTip">
        <string>Символ, кратность которого нужно учитывать</string>
       </property>
      </widget>
     </item>
     <item row="1" column="4" rowspan="11">
      <widget class="QPlainTextEdit" name="teRegEx">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Окно ввода регулярного выражения.&lt;/p&gt;&lt;p&gt;Допустимые операторы:&lt;/p&gt;&lt;p&gt;+ - или&lt;/p&gt;&lt;p&gt;* - повторение от 0 до бесконечности&lt;/p&gt;&lt;p&gt;() - группировка&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="plainText">
        <string>(ab+bc)*</string>
       </property>
       <property name="placeholderText">
        <string/>
       </property>
      </widget>
     </item>
     <item row="9" column="1" rowspan="2">
      <widget class="QSpinBox" name="sbFrom">
       <property name="toolTip">
        <string>Минимальная длина цепочек на вывод</string>
       </property>
       <property name="maximum">
        <number>10</number>
       </property>
      </widget>
     </item>
     <item row="12" column="0" colspan="4">
      <widget class="QTreeWidget" name="treeWidget">
       <property name="uniformRowHeights">
        <bool>true</bool>
       </property>
       <property name="allColumnsShowFocus">
        <bool>true</bool>
       </property>
       <column>
        <property name="text">
         <string notr="true">1</string>
        </property>
       </column>
      </widget>
     </item>
     <item row="12" column="4">
      <widget class="QListWidget" name="lwOutput"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="mFile">
    <property name="title">
     <string>Файл</string>
    </property>
    <addaction name="aSave"/>
   </widget>
   <widget class="QMenu" name="mAbout">
    <property name="title">
     <string>Справка</string>
    </property>
    <addaction name="aTheme"/>
    <addaction name="aAuthor"/>
   </widget>
   <addaction name="mFile"/>
   <addaction name="mAbout"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="aOpen">
   <property name="text">
    <string>Открыть</string>
   </property>
   <property name="toolTip">
    <string>Открыть</string>
   </property>
  </action>
  <action name="aSave">
   <property name="text">
    <string>Сохранить</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="aTheme">
   <property name="text">
    <string>О Теме</string>
   </property>
  </action>
  <action name="aAuthor">
   <property name="text">
    <string>Об Авторе</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>leTerm</tabstop>
  <tabstop>cbRepeat</tabstop>
  <tabstop>sbMul</tabstop>
  <tabstop>leSubString</tabstop>
  <tabstop>pbCreateReg</tabstop>
  <tabstop>sbFrom</tabstop>
  <tabstop>sbTo</tabstop>
  <tabstop>pbOutput</tabstop>
  <tabstop>teRegEx</tabstop>
  <tabstop>treeWidget</tabstop>
  <tabstop>lwOutput</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
